src/runtime/mutex.cpp

```diff
@@ -4,6 +4,7 @@ Released under Apache 2.0 license as described in the file LICENSE.
 
 Authors: Gabriel Ebner, Henrik Böving
 */
+#include <shared_mutex>
```

Comment:
I think the way you're supposed to do this is put this import in `runtime/thread.h` and then provide a dummy implementation if threading support is disabled. Though to be honest I have no idea if building Lean without multithreading is supported/working at this point.